src/idpi/operators/regrid.py

```diff
@@ -0,0 +1,184 @@
+"""Regridding operator."""
+
+# Standard library
+import dataclasses as dc
+import typing
+
+# Third-party
+import numpy as np
+import xarray as xr
+from rasterio import transform
+from rasterio import warp
+from rasterio.crs import CRS
+
+Resampling: typing.TypeAlias = warp.Resampling
+
+CRS_ALIASES = {
+    "geolatlon": "epsg:4326",
+    "swiss": "epsg:21781",
+    "swiss03": "epsg:21781",
+    "swiss95": "epsg:2056",
+    "boaga-west": "epsg:3003",
+    "boaga-east": "epsg:3004",
+}
+
+
+def _get_crs(geo):
+    if geo["gridType"] != "rotated_ll":
+        raise NotImplementedError("Unsupported grid type")
+
+    lon = geo["longitudeOfSouthernPoleInDegrees"]
+    lat = -1 * geo["latitudeOfSouthernPoleInDegrees"]
+
+    return CRS.from_string(
+        f"+proj=ob_tran +o_proj=longlat +o_lat_p={lat} +lon_0={lon} +datum=WGS84"
+    )
+
+
+def _relimit(longitude: float) -> float:
+    if longitude > 180:
+        return longitude - 360
+    return longitude
```

Comment:
Is this an incomplete renormalization of arbitrary angles to `(-pi, +pi]`?
Or is this renormalizing angles in `[0, 2*pi)` to `[-pi, +pi)`?
This function needs documentation on pre and post conditions, which explicitly mentions which bounds (upper/lower) are included.

Reply:
Because the domain is centered on the origin the xmin is usually negative but sometimes it's given in the [0, 360] range. In that case, xmin > xmax and some of the formulas breakdown. This function is supposed to make sure that xmin and xmax are compatible.

Reply:
I understand, and this would replace the helper function with a general purpose function, that normalizes any angle to the interval [-180, +180). If you like it, add it, if you don't, ignore it.
```
import math

def normalize_angle(angle: float) -> float:
    return math.fmod(angle + 180, 360) - 180
```

Reply:
Thanks for the suggestion, I'll use the version from numpy since that's readily available in the current module.

Reply:
even better